src-backend/api/views.py

```diff
@@ -13,6 +17,18 @@ def get_queryset(self):
         user = self.request.user
         return models.Procedure.objects.filter(owner=user)
 
+    @detail_route(methods=['get'])
+    def generate(self, request, pk=None):
+        try:
+            protocol = ProtocolBuilder.generate(request.user, pk)
+        except ValueError:
+            return Response('Invalid procedure', status=status.HTTP_404_NOT_FOUND)
```

Comment:
Should use this instead https://docs.djangoproject.com/en/1.8/topics/http/views/#the-http404-exception